src/Raven.Client/Documents/EntityToBlittable.cs

```diff
@@ -15,32 +16,70 @@
 using Sparrow.Json;
 using Sparrow.Json.Parsing;
 using DocumentInfo = Raven.Client.Documents.InMemoryDocumentSessionOperations.DocumentInfo;
+using System.Text;
 
 namespace Raven.Client.Documents
 {
     public class EntityToBlittable
     {
-        private readonly IDocumentStore documentStore;
-        private readonly JsonOperationContext _context;
+        private readonly InMemoryDocumentSessionOperations _session;
 
-        /// <summary>
-        /// All the listeners for this session
-        /// </summary>
-        public EntityToBlittable(IDocumentStore documentStore, JsonOperationContext context)
+        public MemoryStream OperationsMemoryStream
         {
-            this.documentStore = documentStore;
-            _context = context;
+            get
+            {
+                if (_stream == null)
+                {
+                    _stream = new MemoryStream(4 * 1024);
+                }
+                return _stream;
+            }
         }
-        public readonly Dictionary<object, Dictionary<string, JToken>> MissingDictionary = new Dictionary<object, Dictionary<string, JToken>>(ObjectReferenceEqualityComparer<object>.Default);
 
-        public BlittableJsonReaderObject ConvertEntityToBlittable(string id, object entity, DocumentInfo documentInfo)
+        private MemoryStream _stream;
+
+
+        public StreamReader _streamReader;
+
+        public StreamReader StreamReader
         {
-           return GetObjectAsBlittable(entity, documentInfo, _context); 
+            get
+            {
+                if (_streamReader == null)
```

Comment:
You are setting a lot of variables to the defaults, why? It would be easier to do `new StreamWriter()`